runtime/oti/jvmtiInternal.h

```diff
@@ -489,7 +481,9 @@ typedef struct jvmtiGcp_translation {
 #if JAVA_SPEC_VERSION >= 19
 #define ENSURE_JTHREAD_NOT_VIRTUAL(vmThread, jthrd, error) \
 	do { \
-		if (IS_VIRTUAL_THREAD((vmThread), J9_JNI_UNWRAP_REFERENCE(jthrd))) { \
+		if ((NULL != (jthrd)) \
```

Comment:
Unless I'm misreading, this does not handle the case of a NULL input parameter where the implied current thread is virtual.

Given that you've moved the NULL check into `getVMThread`, perhaps we should just add another parameter `allowVirtual` and keep all the tests in one place to avoid repeatedly NULL checking and potentially fetching the current thread.

Reply:
In addition to `allowVirtual` (bitfield), we will also have to pass the type of error. Different methods throw different errors vthreads: JVMTI_ERROR_UNSUPPORTED_OPERATION and JVMTI_ERROR_OPAQUE_FRAME.